src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs

```diff
@@ -348,19 +348,21 @@ public override void Transform(RavenEtlItem item, EtlStatsScope stats, EtlProces
             }
         }
 
-        private bool ShouldLoadTimeSeriesWithDoc(RavenEtlItem item, EtlProcessState state)
+        private bool ShouldLoadTimeSeriesWithDoc(Document doc, EtlProcessState state)
         {
             //If an Etag of time-series is lower then its document Etag then replication can send the time-series before the document.
             //In this situation Etl process will skip the time-series, mark it as skipped and will send all of it here with the document
-            return state.SkippedTimeSeriesDocs != null && state.SkippedTimeSeriesDocs.Remove(item.DocumentId);
+            return state.SkippedTimeSeriesDocs != null && state.SkippedTimeSeriesDocs.Remove(doc.Id);
         }
 
-        private bool ShouldFilterOutDeletion(RavenEtlItem item)
+        private bool ShouldFilterOutDeletion(Document doc, bool isDelete)
         {
             if (_script.HasDeleteDocumentsBehaviors)
             {
-                var collection = item.Collection ?? item.CollectionFromMetadata;
-                var documentId = item.DocumentId;
+                doc.TryGetMetadata(out var metadata);
+
+                var collection = metadata[Constants.Documents.Metadata.Collection].ToString();
```

Comment:
I have some doubts about this change. Let's talk directly about it.

Reply:
I suppose it's the best way to go. We don't want to do something else - we need the collection name. If the metadata isn't present, these lines should throw NRE for a good reason.